01_task_block.py

```diff
@@ -0,0 +1,177 @@
+License: BSD (3-clause)
+"""
+import matplotlib.pyplot as plt
+
+from mne.io import read_raw_fif
+from mne import events_from_annotations, concatenate_raws, open_report
+
+# All parameters are defined in config.py
+from config import fname, n_jobs, parser, LoggingFormat
+
+# Handle command line arguments
+args = parser.parse_args()
+subject = args.subject
+
+print(LoggingFormat.PURPLE +
+      LoggingFormat.BOLD +
+      'Extracting task blocks for subject %s' % subject +
+      LoggingFormat.END)
+
+###############################################################################
+# 1) Import the output from previous processing step
+input_file = fname.output(subject=subject,
+                          processing_step='raw_files',
+                          file_type='raw.fif')
+raw = read_raw_fif(input_file, preload=True)
+
+# drop status channel
+raw.drop_channels('Status')
+
+###############################################################################
+# 2) Find periods of time in the data with no presented stimuli (i.e., the
+# self-paced breaks)
+
+# relevant events
+ids = {'245': 1,  # end of block
+       '71': 2,   # flanker onset
+       '11': 3,   # target_L_C
+       '12': 4,   # target_R_C
+       '21': 5,   # target_L_I
+       '22': 6,   # target_R_I
+       '101': 7,  # correct_incongruent
+       '102': 8,  # correct_congruent
+       '201': 9,  # incorrect_incongruent
+       '202': 10  # incorrect_congruent
+       }
+
+# extract events
+events = events_from_annotations(raw, event_id=ids)
+
+# cue events
+cue_evs = events[0]
+cue_evs = cue_evs[(cue_evs[:, 2] >= 7) & (cue_evs[:, 2] <= 12)]
+
+# latencies and difference between two consecutive cues
+latencies = cue_evs[:, 0] / raw.info['sfreq']
+diffs = [(y - x) for x, y in zip(latencies, latencies[1:])]
+
+# get first event after a long break (i.e., when the time difference between
+# stimuli is greater than 10 seconds). This should only be the case in between
+# task blocks.
+breaks = [diff for diff in range(len(diffs)) if diffs[diff] > 10]
+print('\n Identified breaks at positions', breaks)
+
+###############################################################################
+# 3) Save start and end points of task blocks
+# subject '041' has more practice trials (two rounds)
+if subject == 41:
+    # start of first block
+    b1s = latencies[breaks[2] + 1] - 2
+    # end of first block
+    b1e = latencies[breaks[3]] + 6
+
+    # start of second block
+    b2s = latencies[breaks[3] + 1] - 2
+    # end of second block
+    b2e = latencies[breaks[4]] + 6
+
+# all other subjects have the same structure
+else:
+    # start of first block
+    b1s = latencies[breaks[0] + 1] - 2
+    # end of first block
+    b1e = latencies[breaks[1]] + 6
+
+    # start of second block
+    b2s = latencies[breaks[1] + 1] - 2
+    # end of second block
+    if len(breaks) > 2:
+        b2e = latencies[breaks[2]] + 6
+    else:
+        b2e = latencies[-1] + 6
```

Comment:
this is just in case any files deviate from the standard format, let's say, if the eeg machine crashed in the middle of the experiment. I think we can delete this part for now.